interprettensor/modules/convolution.py

```diff
@@ -65,14 +73,32 @@ def forward(self, input_tensor):
         self.in_N, self.in_h, self.in_w, self.in_depth = self.input_tensor.get_shape().as_list()
         
         # init weights
-        self.weights_shape = [self.kernel_size, self.kernel_size, self.in_depth, self.output_depth]
+        self.weights_shape = [self.kernel_size, self.kernel_size, self.in_depth/self.groups, self.output_depth]
         self.strides = [1,self.stride_size, self.stride_size,1]
         with tf.variable_scope(self.name):
-            self.weights = variables.weights(self.weights_shape)
-            self.biases = variables.biases(self.output_depth)
+            if self.weights is None:
```

Comment:
Before, I wasn't able to set the weights to pre-trained weights after initialization, so I moved that logic to the point where we directly initialize the weights Variable. I'll take a look at the recent update though.